File: Dictionaries and Sets/1700. Number of Students Unable to Eat Lunch.py
class Solution:
    def countStudents(self, students: List[int], sandwiches: List[int]) -> int:
        ans = len(students)
        c = Counter(students)

        for s in sandwiches:
            if c[s] > 0:
                c[s] -= 1
                ans -= 1
            else:
                return ans
                
        return ans
            
# Time: O(n)
# Space: O(n)

File: Dictionaries and Sets/1700. Number of Students Unable to Eat Lunch/1700. Number of Students Unable to Eat Lunch.js
/**
 * @param {number[]} students
 * @param {number[]} sandwiches
 * @return {number}
 */
var countStudents = function (students, sandwiches) {
  let ans = students.length;

  const counter = { 0: 0, 1: 0 };
  for (const s of students) {
    counter[s]++;
  }

  for (const s of sandwiches) {
    if (counter[s] <= 0) return ans;
    counter[s]--;
    ans--;
  }

  return ans;
};

// Time: O(n)
// Space: O(1)
